calcom/cal.com
# V2.0 - Routing Forms and Shell 2.0 everywhere

## What does this PR do?

_Ignore the number of commits - These are high due to many branches being merged in it._



[Loom Demo](https://www.loom.com/share/643b6c763a5f4f5296c719b4b91310db)
[Demo Showing Shell is not re-rendered](https://www.loom.com/share/99e322e61858459c91dc6b6f02d07040)
Fixes #3794 #3795
It also updates Shell to v2 with the following fixes
- Layout fixes for tablet and mobile screens
- i18n flicker
- Layout fixes in case of child nav items.

I took this opportunity to do following as well:
- Security Hardening for routing forms endpoints.
- Refactored code to share actions across list and single view of Routing Form.



**Environment**: Production

## Type of change



## How should this be tested?


Prerequisite: _Go and install the Routing Forms App first and then you would see Routing Forms link in Main Navigation_
**Tests for V2.0 Routing Forms**
- [ ] Click on Routing Forms and you would see either a zero state or a list of forms. Both should be as per the designs in https://www.figma.com/file/9MOufQNLtdkpnDucmNX10R/%E2%9D%96-Cal-DS-v2.0?node-id=16195%3A132324
- [ ] Go and add a Form and play with adding fields 
- [ ] Go to Routing Tab and Play with Routing. (_I don't think we have documentation on this but should be intutive_). Everything should conform to new designs and work as expected.

**Tests for V2.0 Shell** - Because Routing Forms V2.0 designs are going live with V1, Shell V2.0 deigns has to also go live with V1
- [ ] Test Main Navigation - Should be as per the designs of mobile, tablet and desktop screens,
